scipy/stats/tests/test_morestats.py

```diff
@@ -1987,6 +1987,91 @@ def test_circfuncs_unit8(self):
         assert_allclose(stats.circstd(x, high=180), 20.91551378, rtol=1e-7)
 
 
+class TestQuartileCoeffOfDispersion(object):
+
+    def test_simple(self):
+        arr = [2, 4, 6, 8, 10, 12, 14, 16]
+
+        # Q_low = 4, Q_high = 12 -> dispersion = (12-4)/(12+4) = 8/16 = 0.5
+        assert_equal(stats.quartile_coeff_dispersion(arr), 0.5)
+
+        arr = [1.6, 2.1, 2.3, 2.4, 2.6, 2.9, 2.98, 3]
+
+        # Q_low = 2.1, Q_high = 2.9 -> dispersion = (2.9-2.1)/(2.9+2.1) = 0.8/5 ~= 0.16
+        assert_allclose(stats.quartile_coeff_dispersion(arr), 0.16)
+
+    def test_simple_many(self):
+        # Test different output shapes, axis values
+
+        ref = [2, 4, 6, 8, 10, 12, 14, 16]
+        arr = np.array([ref] * len(ref)).transpose()
+
+        # Same as test_simple_flattened but with arr as repeated columns, expecting len(arr) 0.5 dispersion values
+        assert_array_equal(stats.quartile_coeff_dispersion(arr, axis=0), [0.5] * len(ref))
+        assert_array_equal(stats.quartile_coeff_dispersion(arr, ), 0.5)
+
+        # Aggregate over row, so each row has the same value -> Q2 = 0 -> dispersion = 0
+        assert_array_equal(stats.quartile_coeff_dispersion(arr, axis=1), [0.0] * len(ref))
+
+    def test_q_values(self):
+        # Tests for different q values
+
+        arr = [2, 4, 6, 8, 10, 12, 14, 16]
+
+        # Q_low (0.25) = 4, Q_high (0.5) = 8 -> dispersion = (8-4)/(8+4) = 4/12 = 1/3
+        assert_equal(stats.quartile_coeff_dispersion(arr, q=(0.25, 0.5)), 1/3)
+
+        arr = [1.6, 2.1, 2.3, 2.4, 2.6, 2.9, 2.98, 3]
+        # Q_low (0.5) = 2.4, Q_high (0.75) = 2.9 -> dispersion = (2.9-2.4)/(2.9+2.4) ~= 0.0943
+        assert_allclose(stats.quartile_coeff_dispersion(arr, q=(0.5, 0.75)), (2.9 - 2.4) / (2.9 + 2.4))
+
+    def test_other_interpolations(self):
+        # Test different interpolation types
+
+        arr = [1, 2, 3, 4, 5, 6, 7, 8]
+
+        # Interpolation = 'midpont', Q_low (0.25) = (3+2)/2 = 2.5,
+        # Q_high (0.75) = (7+6)/2 = 6.5 => dispersion = 4/9
+        assert_equal(stats.quartile_coeff_dispersion(arr, interpolation='midpoint'), 4/9)
+        # Interpolation = 'midpont', Q_low (0.25) = 2
+        # Q_high (0.75) = 6 => dispersion = 4/8
+        assert_equal(stats.quartile_coeff_dispersion(arr), 4/8)
```

Comment:
```suggestion
        # Interpolation = 'midpoint', Q_low (0.25) = (3+2)/2 = 2.5,
        # Q_high (0.75) = (7+6)/2 = 6.5 => dispersion = 4/9
        assert_equal(stats.quartile_coeff_dispersion(arr, interpolation='midpoint'), 4/9)
        # Interpolation = 'lower' (default), Q_low (0.25) = 2
        # Q_high (0.75) = 6 => dispersion = 4/8
        assert_equal(stats.quartile_coeff_dispersion(arr), 4/8)
```
Oops, I didn't realize that lower was the default when I wrote the suggestion. Thanks for correcting it.

Reply:
Sorry if I'm forgetting; we might have discussed it before - curious why that is the default?

Reply:
Hmm, I tried to find what might be the best default interpolation.
I took a look on the [wiki](https://en.wikipedia.org/wiki/Quartile_coefficient_of_dispersion) page, and the examples (which were pretty simple) seemed to use `lower` interpolation - therefore I followed.

This is a rather tricky one, the default interpolation value for `numpy.quantile` is actually `linear`. I'd be happy to change it if you suggest we comply with numpy.

Reply:
Interesting... actually those Wikipedia examples don't seem to use any of the available interpolation options, right? It's possible the author intended for there to be more data points, e.g.
```
a = [0, 2, 4, 6, 8, 10, 12, 14, 16]
```
Then no interpolation would be needed because the 25th and 75th percentiles lie right on the data. Do you think they made a mistake?

Reply:
Hmm... actually, the definition of the 1st and 3rd quartiles I'm finding in some places is not identical to 25th and 75th percentiles.

Reply:
Interesting, I wasn't familiar with this one. 
I think I kind of implemented Method 4 from the wiki page, didn't I?

Reply:
Do we want to implement all different methods? I could do it.

Reply:
Using no.quantile you get method 4, yes.
I don't think you need all four, no. I will make a suggesting to the documentation later today, and i think this will be resolved.